framework/include/auxkernels/MaterialStdVectorRealGradientAux.h

```diff
@@ -0,0 +1,57 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#ifndef MATERIALSTDVECTORREALGRADIENTAUX_H
+#define MATERIALSTDVECTORREALGRADIENTAUX_H
+
+// MOOSE includes
+#include "MaterialStdVectorAuxBase.h"
+
+// Forward declarations
+class MaterialStdVectorRealGradientAux;
+
+template<>
+InputParameters validParams<MaterialStdVectorRealGradientAux>();
+
+/**
+ * AuxKernel for outputting a std::vector material-property component to an AuxVariable
+ */
+class MaterialStdVectorRealGradientAux : public MaterialStdVectorAuxBase<RealGradient>
+{
+public:
+
+  /**
+   * Class constructor
+   * @param name AuxKernel name
+   * @param parameters The input parameters for this object
+   */
+  MaterialStdVectorRealGradientAux(const std::string & name, InputParameters parameters);
+
+  /**
+   * Class destructor
```

Comment:
This comment states the obvious, just like the comment for the constructor. I suggest we avoid that.